integration/indexes_command_compat_test.go

```diff
@@ -0,0 +1,486 @@
+			key:            bson.D{},
+			resultType:     emptyResult,
+		},
+		"KeyNotSet": {
+			collectionName: "test",
+			resultType:     emptyResult,
+		},
+		"UniqueFalse": {
+			collectionName: "unique_false",
+			key:            bson.D{{"v", 1}},
+			indexName:      "unique_false",
+			unique:         false,
+		},
+		"UniqueTypeDocument": {
+			collectionName: "test",
+			key:            bson.D{{"v", 1}},
+			indexName:      "test",
+			unique:         bson.D{},
+			resultType:     emptyResult,
+		},
+	} {
+		name, tc := name, tc
+		t.Run(name, func(t *testing.T) {
+			if tc.skip != "" {
+				t.Skip(tc.skip)
+			}
+
+			t.Helper()
+			t.Parallel()
+
+			indexesDoc := bson.D{}
+
+			if tc.key != nil {
+				indexesDoc = append(indexesDoc, bson.E{Key: "key", Value: tc.key})
+			}
+
+			if tc.indexName != nil {
+				indexesDoc = append(indexesDoc, bson.E{"name", tc.indexName})
+			}
+
+			if tc.unique != nil {
+				indexesDoc = append(indexesDoc, bson.E{Key: "unique", Value: tc.unique})
+			}
+
+			var targetRes bson.D
+			targetErr := targetCollection.Database().RunCommand(
+				ctx, bson.D{
+					{"createIndexes", tc.collectionName},
+					{"indexes", bson.A{indexesDoc}},
+				},
+			).Decode(&targetRes)
+
+			var compatRes bson.D
+			compatErr := compatCollection.Database().RunCommand(
+				ctx, bson.D{
+					{"createIndexes", tc.collectionName},
+					{"indexes", bson.A{indexesDoc}},
+				},
+			).Decode(&compatRes)
+
+			if targetErr != nil {
+				t.Logf("Target error: %v", targetErr)
+				t.Logf("Compat error: %v", compatErr)
+
+				// error messages are intentionally not compared
+				AssertMatchesCommandError(t, compatErr, targetErr)
+
+				return
+			}
+			require.NoError(t, compatErr, "compat error; target returned no error")
+
+			if tc.resultType == emptyResult {
+				require.Nil(t, targetRes)
+				require.Nil(t, compatRes)
+			}
+
+			assert.Equal(t, compatRes, targetRes)
+
+			targetCursor, targetErr := targetCollection.Indexes().List(ctx)
+			compatCursor, compatErr := compatCollection.Indexes().List(ctx)
+
+			if targetCursor != nil {
+				defer targetCursor.Close(ctx)
+			}
+			if compatCursor != nil {
+				defer compatCursor.Close(ctx)
+			}
+
+			require.NoError(t, targetErr)
+			require.NoError(t, compatErr)
+
+			targetListRes := FetchAll(t, ctx, targetCursor)
+			compatListRes := FetchAll(t, ctx, compatCursor)
+
+			assert.Equal(t, compatListRes, targetListRes)
+
+			targetSpec, targetErr := targetCollection.Indexes().ListSpecifications(ctx)
+			compatSpec, compatErr := compatCollection.Indexes().ListSpecifications(ctx)
+
+			require.NoError(t, compatErr)
+			require.NoError(t, targetErr)
+
+			assert.Equal(t, compatSpec, targetSpec)
+		})
+	}
+}
+
+// TestCreateIndexesCommandCompatCheckFields check that the response contains response's fields
+// such as numIndexBefore, numIndexAfter, createdCollectionAutomatically
+// contain the correct values.
+func TestCreateIndexesCommandCompatCheckFields(t *testing.T) {
```

Comment:
This test is not table driven test because we want to test a specific sequence of actions to check the responses caused by this sequence:
- createdCollectionAutomatically to be true first time because index creation caused collection creation
- createdCollectionAutomatically to false second time because no new collection was created
- note field to be set because no new indexes were created